content/source/assets/javascripts/application.js

```diff
@@ -15,8 +15,35 @@ document.addEventListener("turbolinks:load", function() {
     // current page. The a.current-page class is added during build by
     // layouts/inner.erb.
     var docsSidebar = $("#docs-sidebar ul.nav.docs-sidenav");
-    docsSidebar.find("ul.nav").addClass("nav-hidden");
-    docsSidebar.find("li").has("a.current-page").addClass("active");
+    docsSidebar.find("li").has(".current-page, .nav-visible").addClass("active");
+    var subNavs = docsSidebar.find("ul").addClass("nav-hidden").parent("li");
+
+    // Make sidebar navs expandable
+    subNavs.addClass("has-subnav");
+    subNavs.on("click", function(e) {
+        if (e.target == this) {
+            $(this).toggleClass("active");
+        }
+        e.stopPropagation();
+    });
+    // For subnavs that don't link to a page, use the whole header as a toggle.
+    docsSidebar.find("a[href^='#']").on("click", function(e) {
```

Comment:
Are there ever any subnav links that are to real in-page anchors? Would this prevent them from working? Instead, maybe we want to use `a[href='#']` to match only empty anchor links.

Reply:
Only as a workaround for bad behavior! They always just link to their own anchor, never to a spot in the content area.

For example, I added "real" anchors to the subnav headers in the [functions list](https://www.terraform.io/docs/configuration/functions.html), because otherwise when you clicked one by mistake, you'd rocket to the top of the page. The backend types sidebar has the same thing. The massive providers don't, which is probably frustrating. 

Since we only added those anchors to suppress a bug, I'm not concerned about preserving them.